sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -164,11 +184,41 @@ AZ_NODISCARD az_result az_iot_hub_client_get_client_id(
 /**
  * @brief Gets the Shared Access clear-text signature.
  * @details The application must obtain a valid clear-text signature using this API, sign it using
- *          HMAC-SHA256 using the Shared Access Key as password then Base64 encode the result.
+ * HMAC-SHA256 using the Shared Access Key as password then Base64 encode the result.
+ *
+ * Use the following APIs when the Shared Access Key is available to the application or stored
+ * within a Hardware Security Module. The APIs are not necessary if X509 Client Certificate
+ * Authentication is used.
+ *
+ * @note This API should be used in conjunction with az_iot_hub_client_sas_get_password().
  *
- * @remark More information available at
+ * @note More information available at
  * https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#security-tokens
  *
+ * A typical flow for using these two API's might look something like the following (note the size
+ * of buffers and non-SDK APIs are for demo purposes only):
+ *
+ * @code
+ * const char* const signature_str = "TST+J9i1F8tE6dLYCtuQcu10u7umGO+aWGqPQhd9AAo=";
+ * az_span signature = AZ_SPAN_FROM_STR(signature_str);
+ * az_iot_hub_client_sas_get_signature(&client, expiration_time_in_seconds, signature, &signature);
+ *
+ * char decoded_sas_key[128];

```

Comment:
nit: follow and model best practices (here/elsewhere)
```suggestion
 * char decoded_sas_key[128] = { 0 };
```